tools/profileBuilder/cmd/list.go

```diff
@@ -96,3 +135,67 @@ func init() {
 	listCmd.Flags().StringP(inputLongName, inputShortName, "", inputDescription)
 	listCmd.MarkFlagRequired(inputLongName)
 }
+
+// for each included package check if there is a newer module major version then the one specified,
+// and if there is update the include accordinly.  returns true if any packages were updated.
+func updateModuleVersions(listDef *model.ListDefinition) (bool, error) {
+	dirty := false
+	for i, entry := range listDef.Include {
+		// "../../services/storage/mgmt/2016-01-01/storage"
+		// "../../services/network/mgmt/2015-06-15/network/v2"
+		target := entry
+		if modinfo.HasVersionSuffix(target) {
+			target = filepath.Dir(target)
+			target = strings.Replace(target, "\\", "/", -1)
+		}
+		modDirs, err := modinfo.GetModuleSubdirs(target)
+		if err != nil {
+			return false, err
+		}
+		if len(modDirs) == 0 {
+			continue
+		}
+		latest := target + "/" + modDirs[len(modDirs)-1]
+		if latest == entry {
+			// already using latest major version
+			continue
+		}
+		listDef.Include[i] = latest
+		dirty = true
+	}
+	return dirty, nil
+}
+
+// returns the last major module version for the specified profile directory (e.g. "v2" etc).
+// if there are no major versions the return value is the empty string.
+func getLatestModVer(profileDir string) (string, error) {
+	subdirs, err := modinfo.GetModuleSubdirs(profileDir)
+	if err != nil {
+		return "", err
+	}
+	modDir := ""
+	if len(subdirs) > 0 {
+		modDir = subdirs[len(subdirs)-1]
+	}
+	return modDir, nil
+}
+
+// generate a go.mod file in the specified module major version directory (e.g. "profiles/foo/v2")
+// the provided module directory is used to calculate the module name.
+func generateGoMod(modDir string) error {
+	err := os.Mkdir(modDir, os.ModeDir|0644)
+	if err != nil {
+		return err
+	}
+	gomod, err := os.Create(filepath.Join(modDir, "go.mod"))
+	if err != nil {
+		return err
+	}
+	defer gomod.Close()
+	mod, err := modinfo.CreateModuleNameFromPath(modDir)
+	if err != nil {
+		return err
+	}
+	_, err = fmt.Fprintf(gomod, "module %s\n\ngo 1.12\n", mod)
```

Comment:
>ngo 1.12 [](start = 42, length = 8)

can we extract this in a constant somewhere, seems like it might get forgotten if at one point it needs to be updated